src/Components/Web.JS/src/Services/NavigationManager.ts

```diff
@@ -76,8 +85,18 @@ export function attachToEventDelegator(eventDelegator: EventDelegator): void {
     const anchorTarget = findAnchorTarget(event);
 
     if (anchorTarget && canProcessAnchor(anchorTarget)) {
-      const href = anchorTarget.getAttribute('href')!;
-      const absoluteHref = toAbsoluteUri(href);
+      const anchorHref = anchorTarget.getAttribute('href')!;
+
+      if (anchorHref.startsWith('#') && anchorHref.length > 1) {
```

Comment:
This looks like it would intercept `href="#something"` but not for `href="current/page#something"`.

We should clarify what the intended behavior is in the case of `href="current/page#something"`. Two possibilities:

1. We should treat this as a navigation to `current/page` even though that's already where you are, and *then* scroll to `#something`. This involves going through the .NET-side routing process and possibly notifying the page that it has new parameters (which may be the same as its previous ones), possibly causing it to re-render.
2. Or, we should not treat this as a navigation at all, since we are already at `current/page`, and instead just short-circuit and scroll to `#something`. The .NET side would not receive any notification and hence wouldn't re-render anything.

My guess from seeing the code here is that the implementation in this PR does option (1). Is that the desired design? To be totally honest I think most developers would not expect there to be a difference between navigating to `#something` and `current/page#something` if you are already at `current/page`.

So this is an open design choice and we can choose whatever we want, but I'm guessing option (2) may be more preferable for most developers.

Reply:
I just checked and with regular native `<a>` tags, the behavior of navigating to `current/page#something` is identical to navigating to `#something` (as in, it does not cause a new page load), so we probably should treat the two as identical in the same way.

Reply:
We might need to have some slightly more subtle rule, like:

 * If you navigate to the existing URL where only the hash has been added/changed **and** that anchor is already in the DOM, then we just scroll to it and don't notify the .NET side. That's the closest to native HTML behavior.
 * But, if the anchor isn't already in the DOM, we don't short-circuit the navigation. We treat it as any other navigation and hence notify the .NET side and begin the whole process of waiting for the anchor to be added to the DOM asynchronously as per [this](https://github.com/dotnet/aspnetcore/pull/47320#issuecomment-1486593691).

Otherwise, if we just short-circuit all navigations to `current/page#hash` even when the anchor isn't in the page, that click will never have any effect.